### ISSUE TITLE
Pod Predicate NodeAffinity failed error seen on default-http-backend pod

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#29882]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

- Deploy an azure v1.19.3 cluster in 2.5.1
- Run pre upgrade checks
- Upgrade rancher to 2.5.2-rc5
- Run post upgrade checks.
- Noticed `Pod Predicate NodeAffinity failed error` seen on default-http-backend pod
<img width="1442" alt="Screen Shot 2020-10-31 at 2 00 29 PM" src="https://user-images.githubusercontent.com/26032343/97790144-6b4ff500-1b83-11eb-9d8c-3447228d3803.png">

**Other details that may be helpful:**
This is seen on one of the clusters in the setups
Other clusters in the setup:
AKS - 1.17
EKS - 1.17
Windows 1.16

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.5.2-rc5
- Installation option (single install/HA):

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Azure node driver

- Kubernetes version (use `kubectl version`):

```
1.19.3
```

SURE-3728

**Comment by maggieliu** [comment]
Does not reproduce all the time. Potential upstream issue: https://github.com/rancher/rke2/issues/127#issuecomment-699254134. Moving out of 2.5.2.

**Comment by sowmyav27** [comment]
- See this issue on a HA local cluster on 2.5-head - commit id: `9ee39b9` on `coredns` pod when the nodes used are flatlinux stable-2605.6.0 instances with t3a.medium. 
- The cluster.yml file has these network settings

```
network:
  plugin: calico
  options:
    calico_flex_volume_plugin_dir: /opt/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds
    flannel_backend_type: vxlan
 
services:
  kube-controller:
    extra_args:
      flex-volume-plugin-dir: /opt/kubernetes/kubelet-plugins/volume/exec/
```

**Comment by annablender** [comment]
This seems to be fixed in Kubernetes 1.21, relevant PR https://github.com/kubernetes/kubernetes/pull/99336. QA will test on that version and confirm if it's still an issue.

**Comment by Auston-Ivison-Suse** [comment]
**Setup For Reproduction**
- rancher Version: v2.6.2
- Ha Deploy

**Downstream Clusters**
- azure node drivers
- rke1
- k8s versions: v1.18.20-rancher1-3, v1.19-16-rancher1-3, v1.20.15-rancher1-1, v1.21.9-rancher1-1. 
- Nginx Default Backend within advanced options

**Steps For Reproduction Attempts**
1. Deploy clusters using the Azure node driver, with the above versions of k8s and Nginx Default Backend enabled.
2. Upgrade rancher version from v2.6.2 to 2.6.3

**Results**

There were no node affinity issues. 

**Setup for Validation**
- rancher Version: v2.6.3 upgrade to 2.6-head(
- Ha Deploy

**Downstream Clusters**
- azure node drivers
- rke1
- k8s versions:  v1.19-16-rancher1-3, v1.20.15-rancher1-1, v1.21.9-rancher1-1. 
- Nginx Default Backend within advanced options

**Steps For Validation Attempts**
1. Deploy clusters using the Azure node driver, with the above versions of k8s and Nginx Default Backend enabled.
2. Upgrade rancher version from v2.6.3 to 2.6-head(e55a04c)

**Results**

The clusters came up fine with no errors relating to Node Affinity
